CMake/Runtime/ProcessManifests.cmake

```diff
@@ -133,22 +133,10 @@ function (process_manifests)
         set (${patchu}root "${CMAKE_MATCH_1}")
 
       elseif (PLUGIN${uPATCH}_URI MATCHES "^https?://(.+)$")
-        message (STATUS "[Plugin ${PLUGIN_NAME}] Downloading${spatch}...")
-        file (MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/plugins${upatch}_dls")
-        set (ark${upatch}_path "${CMAKE_BINARY_DIR}/plugins${upatch}_dls/${PLUGIN_NAME}.ark")
-        if (EXISTS "${ark${upatch}_path}")
-          message (STATUS "[Plugin ${PLUGIN_NAME}] Download was cached")
-        else ()
-          file (DOWNLOAD "${CMAKE_MATCH_1}" "${ark${upatch}_path}" STATUS ark${upatch}_dlstatus)
-          list (GET ark${upatch}_dlstatus 0 ark${upatch}_dlstatus_code)
-          if (ark${upatch}_dlstatus_code)
-            list (GET ark${upatch}_dlstatus 1 ark${upatch}_dlstatus_message)
-            file (REMOVE "${ark${upatch}_path}")
-            message (FATAL_ERROR "[Plugin ${PLUGIN_NAME}] Download failed: (${ark${upatch}_dlstatus_code}) ${ark${upatch}_dlstatus_message}")
-          endif ()
-          message (STATUS "[Plugin ${PLUGIN_NAME}] Download complete")
+        cached_download (URL "${CMAKE_MATCH_1}" DEST ark${upatch}_path RESULT_VARIABLE REST)
+        if (REST)
+          message (FATAL_ERROR "${RESULT_VARIABLE}")
```

Comment:
`REST` doesn't sound very meaningful; also please use lowercase names for local variables (which don't leak out of the function)